File: ExamensArbete/a1-main (1)/a1-main/test-files/Ai_Generated.py
import random

class Dice:
    def __init__(self, sides=6):
        self.sides = sides
    
    def roll(self):
        return random.randint(1, self.sides)

class Player:
    def __init__(self, name):
        self.name = name
        self.score = 0
    
    def update_score(self, points):
        self.score += points

def main():
    player1 = Player("Player 1")
    player2 = Player("Player 2")
    dice = Dice()

    for _ in range(10):
        roll1 = dice.roll()
        roll2 = dice.roll()

        if roll1 > roll2:
            player1.update_score(1)
        elif roll1 < roll2:
            player2.update_score(1)
    
    print(f"{player1.name}: {player1.score} points")
    print(f"{player2.name}: {player2.score} points")

if __name__ == "__main__":
    main()

File: ExamensArbete/a1-main (1)/a1-main/test-files/Ai_Generated.js
// JavaScript code
class Dice {
    constructor(sides) {
        this.sides = sides;
    }

    roll() {
        return Math.floor(Math.random() * this.sides) + 1;
    }
}

class Player {
    constructor(name) {
        this.name = name;
        this.score = 0;
    }

    updateScore(points) {
        this.score += points;
    }
}

let player1 = new Player("Player 1");
let player2 = new Player("Player 2");
let dice = new Dice(6);

for (let i = 0; i < 10; i++) {
    let roll1 = dice.roll();
    let roll2 = dice.roll();

    if (roll1 > roll2) {
        player1.updateScore(1);
    } else if (roll1 < roll2) {
        player2.updateScore(1);
    }
}

console.log(player1.name + ": " + player1.score + " points");
console.log(player2.name + ": " + player2.score + " points");
